common/RegexUtil.hpp

```diff
@@ -74,6 +100,8 @@ class RegexListMatcher final
 private:
     std::set<std::string> _allowed;
     std::set<std::string> _denied;
+    /// Memoizes the result of match().
+    mutable std::unordered_map<std::string, bool> _matchedMemo;
```

Comment:
as an aside: if we end up capping at 16 entries anyway, using std::unordered_map over a std::vector might be overkill

Reply:
Agreed. Was thinking of increasing it to something more significant.

On the other hand, in practice we probably get virtually all requests for the very same hostname, so perhaps even vector is overkill and all we need is a single string+bool.